## Практическое задание N2

### Урок 2. Дискретные случайные величины. Закон распределения вероятностей. Биномиальный закон распределения. Распределение Пуассона

In [55]:
from math import factorial

In [56]:
import numpy as np


def combinations(n, k):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))


def bi_rasp(n, k, p):
    return combinations(n, k) * (p ** k) * ((1 - p) ** (n - k))


def puasson(m, l):
    return (l ** m) / factorial(m) * np.e ** (-l)



1. Вероятность того, что стрелок попадет в мишень, выстрелив один раз, равна 0.8. Стрелок выстрелил 100 раз. Найдите вероятность того, что стрелок попадет в цель ровно 85 раз.


In [57]:
C85_100 = combinations(100, 85)
C85_100

253338471349988640

In [58]:
Bp = C85_100 * (0.8) ** 85 * (0.2) ** 15
Bp

0.048061793700746556

In [59]:
Bp = bi_rasp(100, 85, 0.8)
Bp

0.048061793700746355

> Вероятность попадания равна 0.048

2. Вероятность того, что лампочка перегорит в течение первого дня эксплуатации,
равна 0.0004. В жилом комплексе после ремонта в один день
включили 5000 новых лампочек.
Какова вероятность, что ни одна из них не перегорит в первый день?
Какова вероятность, что перегорят ровно две?

> a) Так как вероятность наступления события маленькая (0.0004),
> а событий много (5000) можно применить распределение Пуассона.
>

$$P_m \approx \frac{\lambda^m}{m!}e^{-\lambda}$$,
где $\lambda = np$.

In [60]:
m = 0
p = 0.0004
lamda = 5000 * p
Pm0 = puasson(m, lamda)
Pm0

0.1353352832366127

> Вероятность того что лампочка не перегорит - 0.135 (13,5%)

> б) Какова вероятность, что перегорят ровно две?

In [61]:
m = 2
p = 0.0004
lamda = 5000 * p
Pm2 = puasson(m, lamda)
Pm2

0.2706705664732254

> вероятность что сгорят 2 лампы - 0.27 (27%)

3. Монету подбросили 144 раза. Какова вероятность, что орел выпадет ровно 70 раз?

$$p(X=k) = C_n^kp^kq^{n-k} $$

n=144, k=70, p=0.5, q=0.5

In [62]:
P144_70 = bi_rasp(144, 70, 0.5)
P144_70

0.06281178035144776

> Вероятность выпадения орла - 0.0628 (6,28%)

4. В первом ящике находится 10 мячей, из которых 7 - белые.
Во втором ящике - 11 мячей, из которых 9 белых.
Из каждого ящика вытаскивают случайным образом по два мяча.
Какова вероятность того, что все мячи белые?
Какова вероятность того, что ровно два мяча белые?
Какова вероятность того, что хотя бы один мяч белый?

> a) Какова вероятность того, что все мячи белые?
>
> **1 способ**

- 1 корзина $$C_7^2$$ число благоприятствующих событий
- 1 корзина $$C_{10}^{2}$$ общее число
- 2 корзина $$C_9^2$$ число благоприятствующих событий
- 2 корзина $$C_{11}^{2}$$ общее число


In [63]:
comb = combinations
comb(7, 2) / comb(10, 2) * comb(9, 2) / comb(11, 2)

0.3054545454545455

> **2 способ**

In [64]:
# первая корзина
p1 = 7 / 10 * 6 / 9
# вторая корзина
p2 = 9 / 11 * 8 / 10

p1 * p2

0.30545454545454537

> вероятность что все белые = 0,305 (30,5%)


> в) В первом ящике находится 10 мячей, из которых 7 - белые.
> Во втором ящике - 11 мячей, из которых 9 белых.
> Из каждого ящика вытаскивают случайным образом по два мяча.
> Какова вероятность того, что хотя бы один мяч белый?

> **1 способ**
>
> вероятность будет равна 1 - P(все 4 - Черные)

- 1 корзина $$C_3^2$$ число благоприятствующих событий
- 1 корзина $$C_{10}^{2}$$ общее число
- 2 корзина $$C_2^2$$ число благоприятствующих событий
- 2 корзина $$C_{11}^{2}$$ общее число

In [65]:
1 - comb(3, 2) / comb(10, 2) * comb(2, 2) / comb(11, 2)

0.9987878787878788

> **2 способ**
>

In [66]:
pb_1 = 3 / 10 * 2 / 9
pb_2 = 2 / 11 * 1 / 10
pb = pb_1 * pb_2
1 - pb

0.9987878787878788

> вероятность что хотя бы 1 мяч будет белым = 0.9987 (99,87%)

б) В первом ящике находится 10 мячей, из которых 7 - белые.
Во втором ящике - 11 мячей, из которых 9 белых.
Из каждого ящика вытаскивают случайным образом по два мяча.
Какова вероятность того, что ровно два мяча белые?

> **1 способ**
>
1 корзина 2 белых, 2 корзина 2 черных

1 корзина 2 черных, 2 корзина 2 белых

1 корзина 1 белый, 2 корзина 1 белый

In [67]:
p1 = comb(7, 2) / comb(10, 2) * comb(2, 2) / comb(11, 2)
p1

0.008484848484848486

In [68]:
p2 = comb(3, 2) / comb(10, 2) * comb(9, 2) / comb(11, 2)
p2

0.04363636363636363

In [69]:
p3 = comb(7, 1) * comb(3, 1) / comb(10, 2) * comb(9, 1) * comb(2, 1) / comb(11, 2)
p3

0.15272727272727274

In [70]:
p = p1 + p2 + p3
p

0.20484848484848486

> **2 способ**

In [71]:
# два меча из 1, 0 из 2

a1_2 = 7 / 10 * 6 / 9
a2_0 = 2 / 11 * 1 / 10
pb1 = a1_2 * a2_0
pb1

0.008484848484848482

In [72]:
# 0 из 1, 2 из 2

a1_0 = 3 / 10 * 2 / 9
a2_2 = 9 / 11 * 8 / 10
pb2 = a1_0 * a2_2
pb2

0.04363636363636363

In [73]:
# 1 корзина: 1 белый, 2 черный или 1 черный, 2 белый
pb1_1 = 7 / 10 * 3 / 9
pb1_2 = 3 / 10 * 7 / 9
pb_1 = pb1_1 + pb1_2
pb_1

# 2 корзина: 1 белый, 2 черный или 1 черный, 2 белый
pb2_1 = 9/11 * 2/10
pb2_2 = 2/11 *  9/10
pb_2 = pb2_1 + pb2_2
pb_2

pb3= pb_1*pb_2

In [74]:
# общая сумма
p_ex_white = pb1 + pb2 + pb3
p_ex_white

0.20484848484848484

> вероятность, что ровно два мяча белые - 0.205 (20.5%)
